### ISSUE TITLE
Look to implement Arcane Research

### ISSUE THREAD

**Issue by mlenser** [mlenser/kryx-rpg-issues#50]
https://old.reddit.com/r/UnearthedArcana/comments/fu7rlu/wizard_arcane_research_learn_new_spells_with/

Probably for the wizard only.

**Comment by Marcloure** [comment]
So, that is a way for the wizard to add spells to their spellbook without copying from others? I think that is cool, specially because that makes the spellbook less dependent of the GM.

If that is the case, I think the Alchemist should also be able to discover formulas by themselves.

**Comment by itamarcu** [comment]
The two standard ways for wizards to gain spells are:

- Level up and learn two new spells
- Copy a spell from a spellbook or scroll, requiring 75 sp and 1 hour, per mana - plus having the spell ready to copy

If we implement arcane research, we need to make it useful enough that people might want to use it, but still less useful than directly copying the spell, in my opinion.

My suggestion:

(added to the "Your Spellbook" rules)
> ***Arcane Research***. You can spend time researching a spell you have heard of or are inspired to create. The spell must be from your power source and cost equal to, or less than, your mana limit. If the spell costs 2 or more mana you must know a number of spells from the spell’s theme equal to the spell’s mana cost minus 1. For each mana cost of the spell, the learning process takes 1 day and costs 50 sp, which you spend on experiments, research, and fine inks. After spending the required amount of days and costs, make an Arcana check, with a DC equal to 10 × the spell's mana cost.  If you succeed, you can scribe the spell in your spellbook, spending the usual time and money on it. If you fail, you must repeat this process if you want to keep researching the spell.

Note that with this rule, you can only really learn up to 3-mana spells, and it will be a DC 30 check. With expertise in arcana, +5 INT, and a proficiency bonus of 4, you'd need to roll 19+ on the check, so it should take you an average of 10 attempts (30 days and 500 sp) to research such a spell, which feels fair to me.

I'd also like to extend this to cantrips, but I'm not sure how to phrase that, and would rather wait on this until scribing cantrips into spellbooks is decided upon.

**Comment by mlenser** [comment]
Lots of discussion on Discord: https://discord.com/channels/701706060440010753/709048873351381042/867726554121240586

**Comment by mlenser** [comment]
Mage Arcane Secrets replaces with Arcane Research.

**Comment by mlenser** [comment]
Alchemical Research added to the Alchemist.

**Comment by mlenser** [comment]
- Witchcrafter now has a grimoire instead of a formula book. [#50](https://github.com/mlenser/kryx-rpg-issues/issues/50)
  - Witchcrafter Ritualist feature now adds the two spells to your grimoire. [#50](https://github.com/mlenser/kryx-rpg-issues/issues/50)

**Comment by sjlevin** [comment]
The current implementation of Arcane Research allows researching cantrips for free due to multiplying by the zero mana cost of the spell. You should explicitly say that it does not work for cantrips until you decide on a fix.

**Comment by mlenser** [comment]
Can you open a new issue? It's easier to track things that way.

Question: is it a problem if mages can learn many cantrips? As long as they take as much time and cost as much as 1 mana spells to learn.

Note to self: decrease mana cost to 4 so it is 1 mana for every 2 hours and the rolls are done every 2 hours.
